framework/wait/waitFor.go

```diff
@@ -0,0 +1,45 @@
+package functions
+
+import (
+	"github.com/rancher/rancher/tests/framework/clients/rancher"
+	ranchFrame "github.com/rancher/rancher/tests/framework/pkg/config"
+	"github.com/rancher/tfp-automation/config"
+	waitAction "github.com/rancher/tfp-automation/framework/wait/action"
+	waitState "github.com/rancher/tfp-automation/framework/wait/state"
+)
+
+const (
+	active            = "active"
+	provisioning      = "provisioning"
+	kubernetesUpgrade = "kubernetes-upgrade"
+	scaleUp           = "scale-up"
+	scaleDown         = "scale-down"
+)
+
+// ExecuteAction is a function that will execute the specified module action.
```

Comment:
this function isn't truly executing anything -- its waiting for a specific action to be executed. -- maybe we can clarify the comment here for future reference.

I think the following would be best:

// ExecuteAction is a function that will wait until the specified action has been executed.

Reply:
Resolved